Example/Tests/SwipeHandleGestureTests.swift

```diff
@@ -0,0 +1,97 @@
+import XCTest
+
+@testable
+import SwipePopViewController
+
+class SwipeHandleGestureTests: XCTestCase {
+    var navigationController: UINavigationControllerMock!
+    var viewController: UIViewController!
+    var percentDrivenMock: UIPercentDrivenInteractiveTransitionMock!
+    
+    override func setUp() {
+        super.setUp()
+        let firstViewController = UIViewController()
+        viewController = UIViewController()
+        navigationController = UINavigationControllerMock()
+        percentDrivenMock = UIPercentDrivenInteractiveTransitionMock()
+
+        navigationController.delegate = navigationController
+        navigationController.viewControllers = [firstViewController, viewController]
+        viewController.addSwipeBackGesture()
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        navigationController = nil
+        viewController = nil
+        percentDrivenMock = nil
+    }
+    
+    func testHandleGestureBeganState() {
+        let selfDelegate: UINavigationControllerDelegate = navigationController
+        let pan = PanGestureBeganStateMock()
+        viewController.handlePanGesture(pan)
+        
+        XCTAssertNotNil(viewController.selfNavigationControllerDelegate)
+        XCTAssertTrue(viewController.selfNavigationControllerDelegate === selfDelegate)
+        XCTAssertNotNil(viewController.swipePopNavigationControllerDelegate)
+        XCTAssertTrue(viewController.swipePopNavigationControllerDelegate === viewController.navigationController!.delegate)
```

Comment:
- Use AssertEqual instead of AssertTrue(a === b)
- Add message "nav delegate is replaced by our swipe pop nav delegate"
- One more assert has to be added to check if the new nav delegate is an object of `SwipePopNavigationControllerDelegate`